module/VuFind/src/VuFind/View/Helper/Root/ThemeConfig.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Theme config view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\Cache\Storage\StorageInterface;
+use Laminas\View\Helper\AbstractHelper;
+use VuFindTheme\ThemeInfo;
+
+/**
+ * Theme config view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class ThemeConfig extends AbstractHelper
+{
+    /**
+     * Cache for merged configs
+     *
+     * @var StorageInterface
+     */
+    protected $cache = [];
+
+    /**
+     * ThemeInfo object to access themeConfig
+     *
+     * @var ThemeInfo
+     */
+    protected $themeInfo;
+
+    /**
+     * Constructor
+     *
+     * @param ThemeInfo        $themeInfo ThemeInfo
+     * @param StorageInterface $cache     StorageInterface
+     */
+    public function __construct(ThemeInfo $themeInfo, StorageInterface $cache)
+    {
+        $this->themeInfo = $themeInfo;
+        $this->cache = $cache;
+    }
+
+    /**
+     * Returns config by path
+     *
+     * Examples:
+     * - 'less' => all of less section
+     * - ['less'] => same as above
+     * - ['less', 'active'] => would return LESS active status
+     *
+     * @param string|string[] $path Path to return from theme.config.php
+     *
+     * @return mixed|null
+     */
+    public function __invoke($path = [])
+    {
+        // Ensure path is an array
+        $path = (array)$path;
+
+        $key = array_shift($path);
+
+        $cacheKey = $this->themeInfo->getTheme() . $key;
+        $cached = $this->cache->getItem($cacheKey);
+
+        if ($cached == null) {
+            $cached = $this->themeInfo->getMergedConfig($key, true);
+            $this->cache->setItem($cacheKey, $cached);
+        }
+
+        foreach ($path as $p) {
+            $cached = $cached[$p] ?? null;
```

Comment:
What if it would be done like this:

if ($cached = $cached[$p] ?? false) {
    return $cached[$p]
}

and in the end we could return return []; to always assure that it is an array. Then we could declare type return for the __invoke() function :array. And casting to array could be removed if the parameter $path = [] had array type declared?

Reply:
I think the critical difference between your proposal and the existing code is that existing code will return the LAST match it finds, whereas yours would return the FIRST match. Of course, if that matters, we could do an array_reverse in combination with your proposal to retain the existing behavior. I haven't taken the time to analyze what's going on here, so I'm not sure if it really makes a difference... but I think that's the next detail to think about at any rate.

Reply:
Yea, it can be crucial difference if it matters :) Just a thought tho!

Reply:
What @LuomaJuha proposed would work if we were to flatten the hierarchy of the config when we load it. That might speed things up.

Currently:
```php
$cached = [
    'a' => [
        'b' => 'magic',
    ],
];
```

Flattened
```php
$cached = [
    'a' => ['b' => 'magic'],
    'b' => 'magic',
];
```

Reply:
Ahh, I understand now -- this isn't about searching a series of paths for possible settings, it's about descending down a tree according to a specific set of branches. I don't think flattening things would make that better, honestly.

Reply:
I wonder if this would be more clear if at some point we replaced the `$cached` variable with `$nextNode` or something like that -- because we first retrieve our tree from the cache, and then we navigate it... and once we start navigating it, a different variable name might make it a little more clear what we are doing.